src/cluster/brokerPool.js

```diff
@@ -143,16 +145,32 @@ module.exports = class BrokerPool {
 
   /**
    * @public
-   * @param {Array<String>} topics
+   * @param {Array<String>} topics topics that minimally should be known in the metadata afterwards
    * @returns {Promise<null>}
    */
   async refreshMetadata(topics) {
+    const getTargetTopics = topicMetadata => {
+      if (!topicMetadata) {
+        return topics
+      }
+      const targetTopics = topicMetadata.map(({ topic }) => topic)
+      return topics.reduce(
+        (result, topic) => (result.includes(topic) ? result : result.concat(topic)),
```

Comment:
Shouldn't this be `result.push(topic)` since topic is a string?

Reply:
Well, no. The result of `result.push(topic)` would be the new length of `result`, but we need the actual result here. But, given that I quite often mix up `push` and `concat`, and this code shouldn't be particularly performance critical either, I rewrote it slightly in https://github.com/tulios/kafkajs/pull/977/commits/6aee9ac9be509690019e66729732473495bf2e54 to be more "obvious":

```js
return topics.reduce(
  (result, topic) => (result.includes(topic) ? result : [...result, topic]),
  targetTopics
)
```

Reply:
[From the MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat):

> `const new_array = old_array.concat([value1[, value2[, ...[, valueN]]]])`
> 
> **ValueN**
> Arrays and/or values to concatenate into a new array [...]

I had no idea Array.concat could push elements onto the array if the value is not itself an array.